show/bgp_common.py

```diff
@@ -0,0 +1,394 @@
+                                combined_route[route][j]['internalNextHopActiveNum'] + len(additional_nh_l)
+                            while len(additional_nh_l):
+                                a_nh = additional_nh_l.pop()
+                                combined_route[route][j]['nexthops'].append(a_nh)
+                        # the nexhops merged, no need to add the new_info
+                        skip_this_new_info = True
+                        break
+            if not proto_matched or not skip_this_new_info:
+                # This new_info is unique and should be added to the route
+                combined_route[route].append(new_info)
+    else:
+        combined_route[route] = new_info_l
+
+def process_route_info(route_info, device, filter_back_end, print_ns_str, asic_cnt, ns_str, combined_route, back_end_intf_set):
+    new_route = {}
+    for route, info in route_info.items():
+        new_info_l = []
+        new_info_cnt = 0
+        while len(info):
+            new_info = info.pop()
+            new_nhop_l = []
+            del_cnt = 0
+            while len(new_info['nexthops']):
+                nh = new_info['nexthops'].pop()
+                if filter_back_end and back_end_intf_set != None and "interfaceName" in nh:
+                    if nh['interfaceName'] in back_end_intf_set:
+                        del_cnt += 1
+                    else:
+                        new_nhop_l.append(copy.deepcopy(nh))
+                else:
+                    new_nhop_l.append(copy.deepcopy(nh))
+            # use the new filtered nhop list if it is not empty. if empty nexthop , this route is filtered out completely
+            if len(new_nhop_l) > 0:
+                new_info['nexthops'] = copy.deepcopy(new_nhop_l)
+                new_info_cnt += 1
+                # in case there are any nexthop that were deleted, we will need to adjust the nexhopt counts as well
+                if del_cnt > 0:
+                    internalNextHopNum = new_info['internalNextHopNum'] - del_cnt
+                    new_info['internalNextHopNum'] = internalNextHopNum
+                    internalNextHopActiveNum = new_info['internalNextHopActiveNum'] - del_cnt
+                    new_info['internalNextHopActiveNum'] = internalNextHopActiveNum
+                new_info_l.append(copy.deepcopy(new_info))
+        if new_info_cnt:
+            if asic_cnt > 1:
+                if filter_back_end:
+                    merge_to_combined_route(combined_route, route, new_info_l)
+                else:
+                    new_route[route] = copy.deepcopy(new_info_l)
+            else:
+                new_route[route] = copy.deepcopy(new_info_l)
+    if new_route:
+        if print_ns_str:
+            combined_route['{}'.format(ns_str)] = copy.deepcopy(new_route)
+        else:
+            combined_route.update(copy.deepcopy(new_route))
+
+def print_show_ip_route_hdr():
+    # This prints out the show ip route header based on FRR 7.2 version.
```

Comment:
if you are using frr, why not use its json output. then you do not need to parse anything.

Reply:
@lguohan Yes. I am already getting the json output from FRR of each namespace.  The parsing is needed if the display option is for front-end where user does not want to see the routes using nexthops that are associated with back-end port/LAG.  Also, merging display is required where the same route may have different nexhop depends on which ASIC was obtained and for display option "front-end" we will need to merge these type of routes from all namespaces and present it to the user.